### PR TITLE
Remove dependency on `num_enum`.

### PR DESCRIPTION
**Pull request by rbartlensky**
Hello! 

I am using `libbpf-rs` in one of my personal projects and I noticed that I was pulling in `toml-edit` via `libbpf-rs` which accounted for 1.3 seconds of my total build time. With the help of `cargo tree` I realized that `num_enum` is pulling in that dependency (and many others!) and I decided to try to remove it from `libbpf-rs` completely. Let me know what you think!

`cargo udeps` also noticed some extra things that can be removed, so I went ahead and removed those.

### CODE REVIEW

**Review by danielocfb — COMMENTED** [review]
Thanks for the proposal! I am in favor of removing these dependencies, I am just not sure about the current implementation hardcoding the values. Can we instead rely on the symbolic values, ala:
```patch
--- libbpf-rs/src/btf/mod.rs
+++ libbpf-rs/src/btf/mod.rs
@@ -96,7 +96,7 @@ impl TryFrom<u32> for BtfKind {
         use BtfKind::*;

         Ok(match value {
-            0 => Void,
+            x if x == Void as u32 => Void,
```

File: libbpf-rs/src/map.rs [hunk 1]
```diff
@@ -1066,6 +1060,55 @@ impl MapType {
...
+            23 => Stack,
+            24 => SkStorage,
+            25 => DevmapHash,
+            26 => StructOps,
+            27 => RingBuf,
+            28 => InodeStorage,
+            29 => TaskStorage,
+            30 => BloomFilter,
+            31 => UserRingBuf,
+            u32::MAX => Unknown,
```
Comment (danielocfb) [inline libbpf-rs/src/map.rs:1100]: Can we remove this case? What purpose does it serve?

**Comment by rbartlensky** [comment]
> Thanks for the proposal! I am in favor of removing these dependencies, I am just not sure about the current implementation hardcoding the values. Can we instead rely on the symbolic values, ala:
> 
> ```diff
> --- libbpf-rs/src/btf/mod.rs
> +++ libbpf-rs/src/btf/mod.rs
> @@ -96,7 +96,7 @@ impl TryFrom<u32> for BtfKind {
>          use BtfKind::*;
> 
>          Ok(match value {
> -            0 => Void,
> +            x if x == Void as u32 => Void,
> ```

I think that's even better! I made the changes suggested.

File: libbpf-rs/src/map.rs (hunk 1 above)
Comment (rbartlensky) [reply libbpf-rs/src/map.rs]: consistency with the "old" implementation, but to be fair it's useless, so I removed it

**Comment by rbartlensky** [comment]
@danielocfb Thank you for taking the time to review! When you're happy with the PR I will squash the fixups away.

**Review by danielocfb — APPROVED** [review]
The changes look great now! Feel free to squash and we can go ahead with the merge. Thanks!

**Pull request merged**